### PR TITLE
p2p: Refactor, move all header verification into net, without changing behavior

### PR DESCRIPTION
**Pull request by troygiorshev**
This is an alternative approach to #15197.  It is a direct extension of #2.

### CODE REVIEW

**Comment by jnewbery** [comment]
You can remove https://github.com/troygiorshev/bitcoin/pull/3/files#diff-eff7adeaec73a769788bb78858815c91R3539-R3546 now that net_processing has no knowledge of the message header structure.

**Comment by troygiorshev** [comment]
> 
> 
> You can remove https://github.com/troygiorshev/bitcoin/pull/3/files#diff-eff7adeaec73a769788bb78858815c91R3539-R3546 now that net_processing has no knowledge of the message header structure.

Nice catch, removed.

File: src/net.h [hunk 1]
```diff
@@ -642,13 +640,14 @@ class TransportDeserializer {
     // read and deserialize data
     virtual int Read(const char *data, unsigned int bytes) = 0;
     // decomposes a message from the context
-    virtual CNetMessage GetMessage(const CMessageHeader::MessageStartChars& message_start, int64_t time) = 0;
+    virtual boost::optional<CNetMessage> GetMessage(int64_t time) = 0;
```
Comment (jonasschnelli) [inline src/net.h]: it should probably use our `Optional` substitute (bridge).

**Comment by jnewbery** [comment]
I suggest you close this now that there's a PR in the upstream repo

**Comment by troygiorshev** [comment]
This is now [bitcoin#19107](https://github.com/bitcoin/bitcoin/pull/19107), closing

**Pull request closed without merging**